### PR TITLE
Allow accounts to be re-activated from the admin APIs

### LINKED ISSUE #7220: Reactivate an account
### Description

Deactivating an account works but trying to reactivate an account does not.
As far as I can tell from synapse code there is no handler for account reactivation.
https://github.com/matrix-org/synapse/blob/master/synapse/rest/admin/users.py#L229
If this is not implemented, wouldn't it be good to point it out in docs? https://github.com/matrix-org/synapse/blob/master/docs/admin_api/user_admin_api.rst

### Steps to reproduce

- deactivate an account using rest api
- account is properly deactivated
- try and reactivate the account

### Version information

- **Homeserver**: 1.12.3+bionic1

- **Install method**: apt

- **Platform**: ubuntu 18.04

### PR DESCRIPTION
**Pull request by clokep**
Fixes #7220

This allows accounts to be re-activated via the admin API. There is no effort to re-join rooms, etc. it simply sets the flag on the database to set the account as active. The API requires that the password is set as well, to allow the account to log back in.

This updates the `PUT /_synapse/admin/v2/users/<user_id>` endpoint, but there's also a `POST /_synapse/admin/v1/deactivate/<user_id>` account. We can add a `POST /_synapse/admin/v1/activate/<user_id>` endpoint, but I don't really see the point, it would essentially be identical (minus you don't pass a `"deactivated": false` in the JSON).

### CODE REVIEW

File: docs/admin_api/user_admin_api.rst [hunk 1]
```diff
@@ -95,6 +95,9 @@ Body parameters:
 
 If the user already exists then optional parameters default to the current value.
 
+In order to re-activate an account ``deactivated`` must be set to ``true`` and a new
```
Comment (richvdh) [inline docs/admin_api/user_admin_api.rst:98]: false, surely?.

Also, presumably the password is only required for non-sso  instances?

File: docs/admin_api/user_admin_api.rst [hunk 2]
```diff
@@ -95,6 +95,9 @@ Body parameters:
```
Comment (richvdh) [inline docs/admin_api/user_admin_api.rst:95]: github won't let me comment on it, but on the line above:

> - ``deactivated``, optional, defaults to ``false``.

should be:

> - ``deactivated``, optional. If unspecified, deactivation state will be left unchanged on existing accounts and set to ``false`` for new accounts.

?

Comment (clokep) [reply docs/admin_api/user_admin_api.rst:95]: Makes sense, yeah. 👍

File: docs/admin_api/user_admin_api.rst (hunk 1 above)
Comment (clokep) [reply docs/admin_api/user_admin_api.rst:98]: 👍

**Comment by clokep** [comment]
Thanks @richvdh! I made the changes to the documentation.

**Pull request merged**